gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -896,11 +914,46 @@ def create(self, mechanism, mechanism_synopsis):
             is_deleted = 0
         )
 
+        # Insert the mechanism evidence
+        if mechanism_support == "evidence":
+            # for each publication (pmid) there is one or more evidence values
+            for evidence in mechanism_evidence:
+                publication_obj = None
+
+                try:
+                    publication_obj = Publication.objects.get(pmid=evidence["pmid"])
+
+                except Publication.DoesNotExist:
+                    raise serializers.ValidationError({"message": f"Could not find publication for PMID '{evidence['pmid']}'"})
+
+                # Get the evidence values
+                for evidence_type in evidence["evidence_types"]:
+                    # type can be: function, rescue, functional alteration or models
+                    type = evidence_type["primary_type"].replace(" ", "_")
+                    values = evidence_type["secondary_type"]
+
+                    # Values are stored in attrib table
+                    for v in values:
+                        try:
+                            evidence_value = Attrib.objects.get(
+                                value = v.lower(),
+                                type__code = f"mechanism_evidence_{type.lower()}"
+                            )
+                        except Attrib.DoesNotExist:
+                            raise serializers.ValidationError({"message": f"Invalid mechanism evidence value '{v.lower()}'"})
+
+                        lgd_mechanism_evidence = LGDMolecularMechanismEvidence.objects.create(
+                            molecular_mechanism = lgd_mechanism,
+                            evidence = evidence_value,
+                            publication = publication_obj,
+                            is_deleted = 0
+                        )
```

Comment:
better to have this code inside the above try block

Reply:
It doesn't need to be inside.
If it finds the `evidence_value` then the next step is to run line 945.